Doc/library/pty.rst

```diff
@@ -45,9 +45,12 @@ The :mod:`pty` module defines the following functions:
    process's standard io. This is often used to baffle programs which insist on
    reading from the controlling terminal.
 
-   The functions *master_read* and *stdin_read* should be functions which read from
-   a file descriptor. The defaults try to read 1024 bytes each time they are
-   called.
+   The functions *master_read* and *stdin_read* are passed a file
+   descriptor which they should read from, and should return a byte
+   string. The default implementation for both functions will read up
+   to 1024 bytes each time the function is called. Returning an empty
+   byte string is interpreted as an end of file (EOF) condition, and
+   that *_read* function will no longer be called.
```

Comment:
"_read" is an implementation and a private function, it should not be mentioned in the doc.

"and that read function will no longer be called." would be good enough and understandable, no?

Reply:
Maybe:

> and those *_read functions will no longer be called.

?

Reply:
Hum, I'm not sure what is "that *_read* function". Does it mean "*master_read* or *stdin_read*"? It's confusing.

Well, "that read function" or just "that function" would be less confusing.

Reply:
"and those read functions will no longer be called" EOF can happen on one function, but not the other, no? Or is the EOF "connected" to these 2 functions? Sorry, I don't know well the pty module.

Reply:
What about:

‘‘‘
The callbacks *master_read* and *stdin_read* are passed a file descriptor which they should read from, and should return a byte string. The default implementation for both callbacks will read up to 1024 bytes each time it is called. If a callback returns an empty byte string, this is interpreted as an end-of-file (EOF) condition, and the callback in question will no longer be called.
’’’

IMO the difference between the two callbacks should also be documented. That may answer one of Victor’s questions. _Master_read_ is passed the pseudoterminal’s master file descriptor to read output from the child, and _stdin_read_ is passed 0, to read the parent’s standard input.

To answer another question: by default on Linux EOF only happens when _stdin_read_ is called. _Master_read_ raises EIO instead, which is caught outside the loop. If both callbacks signal EOF, _spawn_ blocks indefinitely, which is described in <https://bugs.python.org/issue26228>.

Reply:
I like your last version @vadmium.  I wasn't actually intending to reference the internal `_read` function, but your phrasing

> the callback in question

communicates that more clearly.  I'll also add something regarding your other notes on the differences between the two callbacks.

Reply:
Thank you to everyone for hammering out the details on the wording.